packages/next/src/server/image-optimizer.ts

```diff
@@ -587,11 +630,30 @@ export async function imageOptimizer(
       'dangerouslyAllowSVG' | 'minimumCacheTTL'
     >
   },
-  isDev: boolean | undefined
-): Promise<{ buffer: Buffer; contentType: string; maxAge: number }> {
+  isDev: boolean | undefined,
+  previousCacheEntry?: IncrementalCacheItem
+): Promise<{
+  buffer: Buffer
+  contentType: string
+  maxAge: number
+  etag: string
+  upstreamEtag: string
+}> {
   const { href, quality, width, mimeType } = paramsResult
-  const upstreamBuffer = imageUpstream.buffer
+  const { buffer: upstreamBuffer, etag: upstreamEtag } = imageUpstream
   const maxAge = getMaxAge(imageUpstream.cacheControl)
+
+  if (shouldUsePreviouslyCachedEntry(imageUpstream, previousCacheEntry)) {
+    // Return the cached optimized image
```

Comment:
```suggestion
```

You can remove this comment.

Alternatively, you can change `shouldUsePreviouslyCachedEntry()` from returning a boolean to returning the result and naming it something like `getPreviouslyCachedOrNull()`